From 73100e1f7085842c3f524b0f84cb5cbdfbcc29db Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:06 -0400
Subject: [infiniband] mlx5: Set QP offsets and parameters for user QPs and not just for kernel QPs

Message-id: <822db61122925dbd7485c0f31e5e576a023087b2.1410425016.git.avadai@redhat.com>
Patchwork-id: 93643
O-Subject: [Patch RHEL7.1 V1 130/197] IB/mlx5: Set QP offsets and parameters for user QPs and not just for kernel QPs
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Haggai Eran <haggaie@mellanox.com>

For user QPs, the creation process does not currently initialize the fields:

 * qp->rq.offset
 * qp->sq.offset
 * qp->sq.wqe_shift

Signed-off-by: Haggai Eran <haggaie@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 48fea837bb2709bda73cd4ae8bbd57cb277f7b90)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx5/qp.c b/drivers/infiniband/hw/mlx5/qp.c
index a89f704..d13ddf1 100644
--- a/drivers/infiniband/hw/mlx5/qp.c
+++ b/drivers/infiniband/hw/mlx5/qp.c
@@ -574,6 +574,10 @@ static int create_user_qp(struct mlx5_ib_dev *dev, struct ib_pd *pd,
  uar_index = uuarn_to_uar_index(&context->uuari, uuarn);
  mlx5_ib_dbg(dev, "uuarn 0x%x, uar_index 0x%x\n", uuarn, uar_index);
 
+ qp->rq.offset = 0;
+ qp->sq.wqe_shift = ilog2(MLX5_SEND_WQE_BB);
+ qp->sq.offset = qp->rq.wqe_cnt << qp->rq.wqe_shift;
+
  err = set_user_buf_size(dev, qp, &ucmd);
  if (err)
   goto err_uuar;
-- 
1.7.1